module/zfs/zvol.c

```diff
@@ -2252,12 +2252,6 @@ zvol_rename_minors_impl(const char *oldname, const char *newname)
 
 		mutex_enter(&zv->zv_state_lock);
 
-		/* If in use, leave alone */
-		if (zv->zv_open_count > 0) {
-			mutex_exit(&zv->zv_state_lock);
-			continue;
-		}
```

Comment:
According to the comment in https://github.com/zfsonlinux/zfs/issues/4344#issue-134463990 this logic was added to make sure the `zv_name` wasn't updated while being accessed by `zvol_ioctl()`.  This was a concern because it was accessed outside the `zv_state_lock`.  Since this is no longer the case, this should be fine.  This does mean the udev symlink name may change out from underneath us,  which is arguably what we want to happen.

```
Function zvol_rename_minors() does not check the reference count before
modifying the zv_name, which is not good since zvol_ioctl(), for instance,
accesses zv_name without holding zvol_state_lock (but with zv_open_count
being positive).
```